### PR TITLE
[Experimental] Product Filters Redesign > Overlay Navigation: Add Block to the Product Filters Block

### LINKED ISSUE #49690: Product Filters Redesign > Parent block: Add overlay navigation block
This is a follow-up to https://github.com/woocommerce/woocommerce/issues/47039

## Expected Behavior:

- We want to create the navigation block that is an inner block of the parent Product Filters block. This is needed when user sets the Overlay mode to `mobile` and `always`.
- The navigation block should have the same settings as the overlay navigation block.

## Reference Image:
Google Chrome -  Block  Product Filters – Figma 2024-07-18 at 11 49 00 AM

### PR DESCRIPTION
**Pull request by thealexandrelara**
### Changes proposed in this Pull Request:

Closes #49690 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

> [!NOTE]  
> **Pre-requisite**
> - If you're using WC Beta Tester plugin, make sure the `experimental-blocks` feature is enabled (Tools > WCA Test Helper > Features).
> - Make sure to reset the Product Filters (Experimental) template part.

1. Navigate to Product Catalog template (Appearance > Editor > Templates > Product Catalog template - or any archive template).
2. Click on "Edit" to open the WordPress editor.
3. In the editor, you should see a block inserter icon or an option to "Add Block". Click on it.
4. Search for "Product Filters (Experimental)" template part in the block inserter. Add it to your content.
5. Open the list view and make sure the Navigation block is added to the Product Filters (Experimental) block.
6. Select the Product Filters (Experimental) block and test the different Overlay settings: Never, Mobile, and Always.
    6.1. When Never is selected, the Navigation block should be hidden in the editor and in the Frontend.
    6.2. When Mobile is selected, the Navigation block should appear only when Mobile is selected.
    6.3. When Always is selected, the Navigation block should always appear. 

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment  Add the Product Filters Overlay Navigation block to the Product Filters block.

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/block-variations.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { BlockVariation } from '@wordpress/blocks';
+
+const variations: BlockVariation[] = [
+	{
+		name: 'product-filters-overlay-navigation-open-trigger',
+		title: __( 'Navigation (Experimental)', 'woocommerce' ),
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/block-variations.tsx]: This should have the same name as the main block (Overlay Navigation), right?

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/block-variations.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Add `isActive` function to all Product Filters Overlay Navigation block variations.
+ * `isActive` function is used to find a variation match from a created
+ *  Block by providing its attributes.
+ */
+variations.forEach( ( variation ) => {
+	// @ts-expect-error: `isActive` is currently typed wrong in `@wordpress/blocks`.
+	variation.isActive = [ 'triggerType' ];
+} );
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/block-variations.tsx]: I don't think we need to distinguish between variations, as they're hidden from the inserter.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/block.json [hunk 3]
```diff
@@ -22,13 +22,21 @@
 		"iconSize": {
 			"type": "string"
 		},
+		"overlayMode": {
+			"type": "string",
+			"default": "never"
+		},
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/block.json:28]: So looks like we need this attribute to hold overlay settings of the `product-filters` block, right? If it's true, then I we can pass the overlay settings from `product-filters` using [block context](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-context/), what do you think?

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx [hunk 4]
```diff
@@ -23,19 +24,198 @@ import { Icon, close } from '@wordpress/icons';
...
+}: {
+	variation: BlockVariationTriggerType;
+} ) => {
+	let label = __( 'Close', 'woocommerce' );
+	if ( variation === 'open-overlay' ) {
+		label = __( 'Filters', 'woocommerce' );
+	}
+
+	return <span>{ label }</span>;
+};
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx:35]: Optional. If we define this and two components below inside the Edit component, we can avoid passing arguments.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx [hunk 5]
```diff
@@ -23,19 +24,198 @@ import { Icon, close } from '@wordpress/icons';
...
+				</>
+			);
+		} else if ( variation === 'close-overlay' ) {
+			return (
+				<>
+					{ overlayNavigationLabel }
+					{ overlayNavigationIcon }
+				</>
+			);
+		}
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx:101]: The return value of these two conditions are the same.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx [hunk 6]
```diff
@@ -23,19 +24,198 @@ import { Icon, close } from '@wordpress/icons';
...
+
+		if (
+			isWithinProductFiltersOverlayTemplatePart &&
+			triggerType === 'open-overlay'
+		) {
+			return true;
+		}
+
+		return false;
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx]: Because the close button always displays, can this be shortened like below?
```
		if ( triggerType === 'close-overlay' ) {
			return false;
		}

		if (
			productFiltersOverlayMode ===
				ProductFiltersBlockOverlayAttribute.NEVER ||
			isWithinProductFiltersOverlayTemplatePart
		) {
			return true;
		}

		return false;
```

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/index.tsx [hunk 7]
```diff
@@ -12,12 +12,15 @@ import { isExperimentalBlocksEnabled } from '@woocommerce/block-settings';
...
 import { Save } from './save';
+import { blockVariations } from './block-variations';
 import './style.scss';
 
 if ( isExperimentalBlocksEnabled() ) {
 	registerBlockType( metadata, {
 		edit: Edit,
 		save: Save,
 		icon: <Icon icon={ closeSquareShadow } />,
+		attributes: metadata.attributes,
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/index.tsx]: Why is this line needed?

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts [hunk 8]
```diff
@@ -1,7 +1,13 @@
+export enum BlockOverlayAttribute {
+	NEVER = 'never',
+	MOBILE = 'mobile',
+	ALWAYS = 'always',
+}
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts]: I guess you're using enum here to have a single source of truth for overlay option. But it also mean that TypeScript is not a superset of Javascript any more, as the value here is used in both type and runtime code. The code still works as expected, but I recommend using union type and constant instead of enum. You can still have a single source of truth by building the type from object, like this:

```ts
const OVERLAY = {
	never: 'never',
	mobile: 'mobile',
	always: 'always',
} as const;

type OverlayOptions = ( typeof OVERLAY )[ keyof typeof OVERLAY ]; // type OverlayOptions = "never" | "mobile" | "always"
```

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php [hunk 9]
```diff
@@ -11,4 +13,80 @@ class ProductFiltersOverlayNavigation extends AbstractBlock {
...
+	/**
+	 * Include and render the block.
+	 *
+	 * @param array    $attributes Block attributes. Default empty array.
+	 * @param string   $content    Block content. Default empty string.
+	 * @param WP_Block $block      Block instance.
+	 * @return string Rendered block type output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
```
Comment (dinhtungdu) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php]: Do we need this? From my test, `get_block_wrapper_attributes()` returns all required style and classes.

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php [hunk 10]
```diff
@@ -11,4 +13,80 @@ class ProductFiltersOverlayNavigation extends AbstractBlock {
...
+		$classes            = $classes_and_styles['classes'] . ' wc-block-product-filters-overlay-navigation';
+		$wrapper_attributes = get_block_wrapper_attributes(
+			array(
+				'class' => trim( $classes ),
+				'style' => trim( $classes_and_styles['style'] ),
```
Comment (dinhtungdu) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php]: This throws a warning:

`Warning: Undefined array key "style" in /Users/dev/workspace/woocommerce/plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php on line 42`

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php [hunk 11]
```diff
@@ -11,4 +13,80 @@ class ProductFiltersOverlayNavigation extends AbstractBlock {
...
+			)
+		);
+
+		if ( 'never' === $attributes['overlayMode'] || ( ! wp_is_mobile() && 'mobile' === $attributes['overlayMode'] ) ) {
+			return null;
+		}
+
+		$html_content = strtr(
```
Comment (dinhtungdu) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php:55]: TIL about strtr, it's way cleaner than sprintf.

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php [hunk 12]
```diff
@@ -11,4 +13,80 @@ class ProductFiltersOverlayNavigation extends AbstractBlock {
...
+	 * Gets the icon to render depending on the triggerType attribute.
+	 *
+	 * @param array $attributes Block attributes.
+	 *
+	 * @return string Label to render on the block
+	 */
+	private function render_icon( $attributes ) {
+		if ( 'open-overlay' === $attributes['triggerType'] ) {
+			return '<svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg" style="width: 16px; height: 16px;"><path d="M10 17.5H14V16H10V17.5ZM6 6V7.5H18V6H6ZM8 12.5H16V11H8V12.5Z" fill="currentColor"></path></svg>';
+		}
```
Comment (dinhtungdu) [inline plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php:79]: Are you planning to support other icon of the open overlay in another PR?

Comment (dinhtungdu) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php:79]: Another approach is including the icon in the saved content. This may remove the need of the PHP `render` method.

**Review by dinhtungdu — CHANGES_REQUESTED** [review]
Thanks for working on this PR. I confirm the instruction is testing well.

Besides the review comments, I have a question related to the settings: I don't see the Icon chooser in the Overlay Navigation block (open trigger variation), are you planning to work on it in this PR or another, probably https://github.com/woocommerce/woocommerce/issues/50273?

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/block-variations.tsx (hunk 1 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/block-variations.tsx]: Done in 1978cefd8000758040e990047648240b7d7e5e9c

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/block-variations.tsx (hunk 2 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/block-variations.tsx]: I removed this in 3abe5fb7351a7d4b12a734dee8a8cce029668517

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/block.json (hunk 3 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/block.json:28]: Great call! Using the block context is much better, I included it in 5bb9f91ebd719ecc610a9c2f33b8b900a87830a6

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx (hunk 5 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx:101]: The order in which each item is displayed is different. Basically, when the variation is `open-overlay`, the icon must appear before the label, and when the variation is `close-overlay`, the label must appear before the icon.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx (hunk 4 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx:35]: If we create a component within another component, React will re-create those components every time the wrapper component re-renders. That way, React isn't able to perform any optimizations since it will consider those inner components as new components.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts (hunk 8 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts]: Thank you for your comment. I believe using `enum` is much more readable and easier to maintain the code than using typeof, keyof, etc. and two separate objects/types to handle that. But I know some part of the community prefers avoiding that, since I don't know what are the guidance regarding to this I changed the code in ee643ce8299135d0acedc10177e800f5316216ee

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php (hunk 12 above)
Comment (thealexandrelara) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php:79]: I believe I'm going to need the logic there to add the interactivity api directives, but if we don't we can think about removing it in the future.

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php (hunk 9 above)
Comment (thealexandrelara) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php]: You're right, I removed it in 13eda9598a568ef3aa134dcd2ff311442b4e934c

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php (hunk 10 above)
Comment (thealexandrelara) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php]: Fixed in 13eda9598a568ef3aa134dcd2ff311442b4e934c

**Comment by thealexandrelara** [comment]
> Besides the review comments, I have a question related to the settings: I don't see the Icon chooser in the Overlay Navigation block (open trigger variation), are you planning to work on it in this PR or another, probably #50273?

Thank you for reviewing this 🙌 ! Looking at the design, I believe we don't have an Icon chooser for the Overlay Navigation block, at least not currently planned, but if we should include one, it would be better to work on a separate PR or as part of #50273.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx (hunk 4 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx:35]: I see, thank you!

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx (hunk 5 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx:101]: Oh, I see the difference now. I'm sorry for overlook and thanks for the clarification.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/block.json [hunk 13]
```diff
@@ -44,7 +44,9 @@
 	},
 	"textdomain": "woocommerce",
 	"usesContext": [ "postId" ],
-	"providesContext": {},
+	"providesContext": {
+		"woocommerce/product-filters-overlay-navigation/overlay": "overlay"
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/block.json]: Can you explain why this needs a namespace and block name?

File: plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php (hunk 12 above)
Comment (dinhtungdu) [reply plugins/woocommerce/src/Blocks/BlockTypes/ProductFiltersOverlayNavigation.php:79]: > I believe I'm going to need the logic there to add the interactivity api directives

We can use HTML API to inject directives.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts [hunk 14]
```diff
@@ -1,7 +1,16 @@
+export const BlockOverlayAttribute = {
+	NEVER: 'never',
+	MOBILE: 'mobile',
+	ALWAYS: 'always',
+} as const;
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts]: As we're talking about separating the type from the runtime, can we move this to a `constants.ts` file like other blocks?

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/index.tsx (hunk 7 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/index.tsx]: Can you look at this? Looks like it was hidden on the main PR page. Also [this one](https://github.com/woocommerce/woocommerce/pull/50186/files#r1707004578).

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx [hunk 15]
```diff
@@ -46,6 +202,10 @@ export const Edit = ( {
 		[]
 	);
 
+	if ( shouldHideBlock() ) {
+		return null;
+	}
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx]: Just want to note the current behavior of the Overlay Navigation block inside Product Filters: When the block is hidden in the inserter, if users select the block in the list view, they can still see the inspector controls generated by block supports 
image

**Review by dinhtungdu — COMMENTED** [review]
Thanks so much for the updates. The PR is still testing well for me. I left some more comments, but overall, this is LGTM!

> Looking at the design, I believe we don't have an Icon chooser for the Overlay Navigation block, at least not currently planned, but if we should include one, it would be better to work on a separate PR or as part of https://github.com/woocommerce/woocommerce/issues/50273.

AFAIU, the open overlay button setting should look like the following image. The difference between open and close is the icon chooser. I agree that we can work on it in #50273.

image

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/block.json (hunk 13 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/block.json]: I'm following the recommendation from [the docs](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-context/). 

> Include a namespace
As seen in the above example, it is recommended that you include a namespace as part of the name of the context key so as to avoid potential conflicts with other plugins or default context values provided by WordPress. The context namespace should be specific to your plugin, and in most cases can be the same as used in the name of the block itself.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx (hunk 15 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx]: Yeah, I believe merchants can edit these blocks even though they are hidden, because if they select "Mobile" or "Always" the block will appear with their customizations.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts (hunk 14 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/types.ts]: Done in 9a04c8d5d95d42fbc7cc73725406fb053a2be60d

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx (hunk 6 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx]: I improved this code in 5e4008b29be27cf6854f1e1ed672b7c523e4ffdb

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/index.tsx (hunk 7 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/index.tsx]: I removed this in 46442b95de64e93294a98caed8986788b81bd39c

**Comment by thealexandrelara** [comment]
Thank you for reviewing it again! I just fixed the issues that you raised. 

> AFAIU, the open overlay button setting should look like the following image. The difference between open and close is the icon chooser. I agree that we can work on it in https://github.com/woocommerce/woocommerce/issues/50273.

Oh I see, yeah, let's work on that in https://github.com/woocommerce/woocommerce/issues/50273

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters/block.json (hunk 13 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters/block.json]: Oh, I wasn't aware of that, thanks. TIL.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx (hunk 15 above)
Comment (dinhtungdu) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx]: Yeah, but users can edit only settings generated by Supports API, our custom inspector settings aren't rendered when the block is hidden.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx [hunk 16]
```diff
@@ -23,19 +23,168 @@ import { Icon, close } from '@wordpress/icons';
...
+			if (
+				productFiltersOverlayMode ===
+				ProductFiltersBlockOverlayAttribute.NEVER
+			) {
+				return true;
+			}
+
+			if ( isWithinProductFiltersTemplatePart ) {
+				return true;
+			}
```
Comment (dinhtungdu) [inline plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx]: This always hides the navigation block when editing the Product Filters template part. Removing it fixes the issue.

**Review by dinhtungdu — CHANGES_REQUESTED** [review]
Thanks for being patient with me and all the updates. Except for a small issue mentioned in the review comment, this PR is looking good to me.

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx (hunk 15 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx]: I got it. I thought you were saying that it should be hidden. I just fixed it, and now the custom settings are being displayed when the block is selected, even though it is hidden a64d44b5c3ac9ce6b98ab4828aa214a93f11c941

File: plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx (hunk 16 above)
Comment (thealexandrelara) [reply plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay-navigation/edit.tsx]: Nice catch! I fixed it in 9d2a73e9bc8511c3be7912c393097cfc2416e016

**Review by dinhtungdu — APPROVED** [review]
Thank you so much for the updates! This is tested well and LGTM!
